OpenSim/Simulation/Model/Appearance.h

```diff
@@ -33,105 +33,157 @@ namespace OpenSim {
-};  // END of class Appearance
-//=============================================================================
+    //=============================================================================
+};  // END of class CurveAppearance
 
 //=============================================================================
-class OSIMSIMULATION_API LineAppearance : public BaseAppearance {
-    OpenSim_DECLARE_CONCRETE_OBJECT(LineAppearance, Object);
+//=============================================================================
+/**
+ * A class that holds the display attributes (Appearance) of an object displayed 
+ * in the OpenSim Visualizer.
+ * 
+ * Appearance contains properties that apply to all geometry.
+ * Geometry that have a surface so that it can be textured will utilize
+ * surface_ppearance, while schematic line drawings (e.g. Arrows, Frames) 
+ * can utilize curve_ppearance which offers thickness. 
+ *
+ * TODO: Add Resolution to Appearance from DecorativeGeometry and 
+ * utilize  for Arrow, Line, and Frame (unused for now)
+ *
+ * @author Ayman Habib
+ * @version 1.0
+ */
+class OSIMSIMULATION_API Appearance : public Object {
+    OpenSim_DECLARE_CONCRETE_OBJECT(Appearance, Object);
 public:
-    //==============================================================================
+    //==========================================================================
     // PROPERTIES
-    //==============================================================================
-    OpenSim_DECLARE_PROPERTY(thickness, double,
-        "The thickness used to visualize a LineGeometry. ");
+    //==========================================================================
+    OpenSim_DECLARE_PROPERTY(visible, bool, 
+        "Flag indicating whether the associated Geometry is visible or hidden.")
+    OpenSim_DECLARE_PROPERTY(opacity, double,
+            "The opacity (0-1) used to display the object. ");
+    OpenSim_DECLARE_PROPERTY(color, SimTK::Vec3,
+        "The color (rgb) used to display the object. ");
+
+    OpenSim_DECLARE_PROPERTY(surface_appearance, SurfaceAppearance,
+        "Visuals applied to surfaces associated with this Appearance. ");
+    OpenSim_DECLARE_PROPERTY(curve_appearance, CurveAppearance,
+        "Visuals applied to curves or line drawings associated with this Appearance. ");
+
 
     //--------------------------------------------------------------------------
     // CONSTRUCTION
     //--------------------------------------------------------------------------
 public:
-    LineAppearance() {
+    Appearance() {
         constructProperties();
     }
-    virtual ~LineAppearance() {};
+    virtual ~Appearance() {};
+
+    DecorativeGeometry::Representation get_representation() const { return (DecorativeGeometry::Representation)get_surface_appearance().get_representation(); }
+
+    void set_representation(DecorativeGeometry::Representation& rep) { upd_surface_appearance().set_representation(rep); }
```

Comment:
There are line length problems here and above. Should not exceed 80 chars. I couldn't see the whole line in the GitHub viewer, but I believe the argument here should be const, or you can just pass it by value (probably better in this case since it is just an int).